packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1011,8 +1011,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
 
         raiseConnectedEvent(this._logger, this, connected, clientId);
 
-        if (connected) {
-            assert(!!clientId);
+        if (connected && clientId) {
```

Comment:
can we increase the precision of this check and ensure if clientId === undefined that deltamanger.storageOnly === true